src/components/material/MaterialAvailabilityText/online/MaterialAvailabilityTextOnline.tsx

```diff
@@ -1,27 +1,63 @@
 import * as React from "react";
-import { useGetV1ProductsIdentifier } from "../../../../core/publizon/publizon";
+import {
+  useGetV1LibraryProfile,
+  useGetV1ProductsIdentifier,
+  useGetV1UserLoans
+} from "../../../../core/publizon/publizon";
 import { useText } from "../../../../core/utils/text";
 import MaterialAvailabilityTextParagraph from "../generic/MaterialAvailabilityTextParagraph";
+import { ManifestationMaterialType } from "../../../../core/utils/types/material-type";
 
 interface MaterialAvailabilityTextOnlineProps {
   isbns: string[];
+  materialType: string;
 }
 
 const MaterialAvailabilityTextOnline: React.FC<
   MaterialAvailabilityTextOnlineProps
-> = ({ isbns }) => {
+> = ({ isbns, materialType }) => {
   const t = useText();
+  const { data: productsData } = useGetV1ProductsIdentifier(isbns[0]);
+  const { data: libraryProfileData } = useGetV1LibraryProfile();
+  const { data: loansData } = useGetV1UserLoans();
+
+  if (!libraryProfileData || !loansData || !productsData) return null;
+
+  const totalEbookLoans = loansData?.userData?.totalEbookLoans;
+  const totalAudioLoans = loansData?.userData?.totalAudioLoans;
+
   const {
-    data: productsData,
-    isLoading: productsIsLoading,
-    isError: productsIsError
-  } = useGetV1ProductsIdentifier(isbns[0]);
+    maxConcurrentEbookLoansPerBorrower,
+    maxConcurrentAudioLoansPerBorrower
+  } = libraryProfileData;
 
-  if (productsIsLoading || productsIsError || !productsData) return null;
+  let availabilityText = "";
 
-  let availabilityText = t("onlineLimitMonthInfoText", {
-    placeholders: { "@count": "X", "@limit": "Y" }
-  });
+  if (
+    materialType === ManifestationMaterialType.ebook &&
+    totalEbookLoans &&
+    maxConcurrentEbookLoansPerBorrower
+  ) {
+    availabilityText = t("onlineLimitMonthEbookInfoText", {
+      placeholders: {
+        "@count": totalEbookLoans,
+        "@limit": maxConcurrentEbookLoansPerBorrower
+      }
+    });
+  }
+
+  if (
+    materialType === ManifestationMaterialType.audioBook &&
+    totalAudioLoans &&
+    maxConcurrentAudioLoansPerBorrower
+  ) {
+    availabilityText = t("onlineLimitMonthAudiobookInfoText", {
```

Comment:
Suggestion for a more function based and, in my opinion, more maintainable solution:
```typescript

type AvailabilityTextMap = {
  [key in ManifestationMaterialType]: {
    text: string;
    count: number;
    limit: number;
  };
};

const getAvaliabilityText = (
  type: ManifestationMaterialType,
  map: AvailabilityTextMap,
  t: UseTextFunction
) => {
  if (!map[type] || !map[type].text || !map[type].count || !map[type].limit) {
    return null;
  }

  const { text, count, limit } = map[type];

  return t(text, {
    placeholders: {
      '@count': count,
      '@limit': limit,
    },
  });
};

export default function App() {
  const t = useText();
  const availabilityTextMap: AvailabilityTextMap = {
    [ManifestationMaterialType.ebook]: {
      text: "onlineLimitMonthEbookInfoText",
      count: totalEbookLoans,
      limit: maxConcurrentEbookLoansPerBorrower,
    },
    [ManifestationMaterialType.audioBook]: {
      text: "onlineLimitMonthAudiobookInfoText",
      count: totalAudioLoans,
      limit: maxConcurrentAudioLoansPerBorrower,
    },
  };
  const text = getAvaliabilityText(
    ManifestationMaterialType.ebook,
    availabilityTextMap,
    t
  );
```